stumpy/stump.py

```diff
@@ -404,26 +427,26 @@ def stump(T_A, m, T_B=None, ignore_trivial=True):
     """
 
     T_A = np.asarray(T_A)
-    core.check_dtype(T_A)
-    core.check_nan(T_A)
-
     if T_A.ndim != 1:  # pragma: no cover
-        raise ValueError(
-            f"T_A is {T_A.ndim}-dimensional and must be 1-dimensional. "
-            "For multidimensional STUMP use `stumpy.mstump` or `stumpy.mstumped`"
-        )
+        raise ValueError(f"T_A is {T_A.ndim}-dimensional and must be 1-dimensional. ")
+    n = T_A.shape[0]
+
+    T_A = T_A.copy()
```

Comment:
Is this copy necessary? Is it to avoid altering the original time series outside of this function?

Reply:
Exactly, because this might produce some undesired results on the user side and it is only one single copy at the beginning